tilequeue/command.py

```diff
@@ -1528,6 +1519,65 @@ def tilequeue_load_tiles_of_interest(cfg, peripherals):
     logger.info('Loading tiles of interest ... done')
 
 
+def tilequeue_tile_status(cfg, peripherals, args):
+    """
+    Report the status of the given tiles in the store, queue and TOI.
+    """
+    logger = make_logger(cfg, 'tile_status')
+
+    # friendly warning to avoid confusion when this command outputs nothing
+    # at all when called with no positional arguments.
+    if not args.coords:
+        logger.warning('No coordinates given on the command line.')
+        return
+
+    # pre-load TOI to avoid having to do it for each coordinate
+    toi = None
+    if peripherals.toi:
+        toi = peripherals.toi.fetch_tiles_of_interest()
+
+    # TODO: make these configurable!
+    tile_format = lookup_format_by_extension('zip')
+    tile_layer = 'all'
+    store = make_store(cfg.store_type, cfg.s3_bucket, cfg)
+
+    for coord_str in args.coords:
+        coord = deserialize_coord(coord_str)
+
+        # input checking! make sure that the coordinate is okay to use in
+        # the rest of the code.
+        if not coord:
+            logger.warning('Could not deserialize %r as coordinate'
+                           % (coord_str,))
+            continue
+
+        if not coord_is_valid(coord):
+            logger.warning('Coordinate is not valid: %r (parsed from %r)'
+                           % (coord, coord_str))
+            continue
+
+        # now we think we probably have a valid coordinate. go look up
+        # whether it exists in various places.
+
+        logger.info("=== %s ===" % (coord_str,))
```

Comment:
:+1:

Thanks! I didn't know that. It makes the code much more readable (bc007b3).